src/main/tray.ts

```diff
@@ -0,0 +1,102 @@
+import { app as rawApp, Menu, Tray } from 'electron'
+import { globalShortcut } from 'electron'
+import * as mainWindow from './windows/main'
+import { ExtendedAppMainProcess } from './types'
+import { getLogger } from '../shared/logger'
+import { appIcon } from './application-constants'
+
+let tray: Tray = null
+let contextMenu: Menu = null
+
+const app = rawApp as ExtendedAppMainProcess
+const log = getLogger('main/tray')
+
+export function updateTrayIcon() {
+  // Don't show a tray icon for mac
+  if (process.platform === 'darwin') return
+
+  // User doesn't want tray icon => destroy it
+  if (app.state.saved.minimizeToTray !== true) {
+    if (tray != null) destroyTrayIcon()
+    return
+  }
+
+  renderTrayIcon()
+}
+
+export function destroyTrayIcon() {
+  log.info('destroy icon tray')
+  tray.destroy()
+  tray = null
+}
+
+export function renderTrayIcon() {
+  if (tray != null) {
+    log.warn('Tray icon not destroyed before render?')
+    destroyTrayIcon()
+  }
+
+  const tx = app.translate
+
+  // Add tray icon
+  log.info('add icon tray')
+  tray = new Tray(appIcon())
+  const win = mainWindow.window
+  contextMenu = Menu.buildFromTemplate([
+    {
+      id: 'open_windows',
+      label: tx('global_menu_file_open_desktop'),
+      type: 'normal',
+      click() {
+        win.show()
+      },
+    },
+    {
+      id: 'reduce_window',
+      label: tx('global_menu_file_reduce_desktop'),
+      type: 'normal',
+      accelerator: 'Escape',
```

Comment:
I don't know what the `accelerator` should do here, I suggest to remove this line